src/makeCalculator.js

```diff
@@ -4,7 +4,51 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  let result = 0;
+
+  function operate(callback, number) {
+    result = callback(result, number);
+
+    return this;
+  }
+
+  function reset() {
+    result = 0;
+
+    return this;
+  }
+
+  function add(a, b) {
+    return a + b;
+  }
+
+  function subtract(a, b) {
+    return a - b;
+  }
+
+  function multiply(a, b) {
+    return a * b;
+  }
+
+  function divide(a, b) {
+    if (b !== 0) {
+      return a / b;
+    } else {
+      throw new Error('Division by zero');
+    }
```

Comment:
The `divide` function correctly checks for division by zero, but it's important to ensure that the `operate` method can handle the exception thrown by this method.